aexpect/remote_door.py

```diff
@@ -104,7 +104,7 @@ def _string_call(function, *args, **kwargs):
             arguments = "%s%s" % (arguments, arg)
         if len(kwargs) > 0 or arg != args[-1]:
             arguments = "%s%s" % (arguments, ", ")
-    ordered_kwargs = list(kwargs.keys())
+    ordered_kwargs = sorted(list(kwargs.keys()))
```

Comment:
Perhaps it'd be nice to add a note here that the sorting is important for selftest purposes. Otherwise someone might optimize this out...

Reply:
Actually the whole function seems unnecessary complex, how about this:

```diff
diff --git a/aexpect/remote_door.py b/aexpect/remote_door.py
index c1c603e..2b5e017 100644
--- a/aexpect/remote_door.py
+++ b/aexpect/remote_door.py
@@ -99,22 +99,15 @@ REMOTE_PYTHON_PATH = "/tmp/utils"
 
 
 def _string_call(function, *args, **kwargs):
-    arguments = ""
-    for arg in args:
+    def arg_to_str(arg):
         if isinstance(arg, str):
-            arguments = "%sr'%s'" % (arguments, arg)
+            return "r'%s'" % arg
         else:
-            arguments = "%s%s" % (arguments, arg)
-        if len(kwargs) > 0 or arg != args[-1]:
-            arguments = "%s%s" % (arguments, ", ")
-    ordered_kwargs = sorted(list(kwargs.keys()))
-    for key in ordered_kwargs:
-        if isinstance(kwargs[key], str):
-            arguments = "%s%s=r'%s'" % (arguments, key, kwargs[key])
-        else:
-            arguments = "%s%s=%s" % (arguments, key, kwargs[key])
-        if key != ordered_kwargs[-1]:
-            arguments = "%s%s" % (arguments, ", ")
+            return "%s" % arg
+    args = tuple(arg_to_str(arg) for arg in args)
+    kwargs = tuple("%s=%s" % (key, arg_to_str(value))
+                   for key, value in sorted(kwargs.items()))
+    arguments = ", ".join(args + kwargs)
     return "result = %s(%s)\n" % (function, arguments)
 
```

Or do you think I should send this separately as it's not related to this change?

Reply:
Absolutely! This little function started simple but then it grew in complexity through the years as we kept adding complications so I guess it reached its point of much needed refactoring. I applied your suggest changes with some credit in the commit message :)